deep-learning-server/data/mnist/mnist_preparation.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import numpy as np
+import cv2
+import os
+import argparse
+
+num_clients = 10
+
+# arguments
+parser = argparse.ArgumentParser(description='Arguments to process the dataset')
+parser.add_argument('--num_clients', type=int, default=1,
+                    help='the number of clients who held frictions of the dataset')
+parser.add_argument('--data_dist', type=str, default='iid',
+                    help="the data distribution among clients [iid] or [non-iid]")
+parser.add_argument('--relat_path_list', type=bool, default=True,
+                    help="which to use relative image path in the loading list")
+args = parser.parse_args()
+
+
+# if MNIST file not exist, then download
+if not os.path.exists("train-images-idx3-ubyte"):
```

Comment:
Currently, only downloads necessary to build the wasm binary are in the Makefile. This wasm example is more extendable and will increasingly support more applications. Thus, maybe for each use case, we put downloads, data preparation, etc in separate files could be better? Otherwise, the main makefile will be filled with things when more applications/use cases are developed.